tales_config.go

```diff
@@ -40,6 +40,14 @@ func WithCache(size int) Option {
 	return func(c *config) { c.CacheSize = size }
 }
 
+// WithParallelDownloads sets the number of parallel chunk downloads.
+func WithParallelDownloads(workers int) Option {
```

Comment:
I prever WithConcurrency(concurrency int) -> default should be runtime.GOMAXPROCS()